### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by danielballan**
I wrote sphinx documentation for pims. I pushed the draft to soft-matter.github.io/pims for your review.

There are no API docs, but I am seeing other lightweight utilities ([1](https://odo.readthedocs.org) [2](multiple-dispatch.readthedocs.org/en/latest/)) skip these as well. Getting them right is at least a day's tedious work, so I'd like to skip them in this first draft at least.

Also, I am anticipating some API-breaking changes. See next issue....

(This PR contains commits from #147, which should be merged first.)

### CODE REVIEW

File: doc/source/bioformats.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+The ``Bioformats`` reader interfaces with Bio-Formats, an open-source Java
+library for reading and writing multidimensional image data, especially from
+file formats used in microscopy.
+
+See the `list of supported formats <https://www.openmicroscopy.org/site/support/bio-formats5.1/supported-formats.html>`_.
+
+Depedencies
+-----------
+To interface with the java library, we use `JPype <https://github.com/originell/jpype>`_, which allows fast and easy access to all java functions. JRE or JDK are not required.
+
```
Comment (caspervdw) [inline doc/source/bioformats.rst]: For installation with ``pip``, make sure to use ``pip install jpype1`` to obtain the correct version (0.6.0 or later).

File: doc/source/opening_files.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+Using Specific Readers
+----------------------
+
+PIMS has several built-in readers. If you don't want to use ``open`` to
+dispatch to them automatically, you can use them directly. For example:
+
+.. code-block:: python
+
+   import pims
+   images = ImageSequence('my_directory/*.png')
```
Comment (caspervdw) [inline doc/source/opening_files.rst]: Could you put `pims.ImageSequence` here?

File: doc/source/opening_files.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+* :doc:`image_sequence`
+* :doc:`tiff_stack`
+* :doc:`video`
+* :doc:`bioformats`
+
+See their individual pages for details.
+
+If you have a file format not yet supported by PIMS, it is easy to define your
+own reader and get PIMS lazy-loading and slicing behavhior "for free."
```
Comment (caspervdw) [inline doc/source/opening_files.rst]: swap . and " ?

File: doc/source/frame.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+PIMS returns images as ``Frame`` objects. Frames can be treated precisely
+the same as numpy arrays.
+
+They are a subclass of ``numpy.ndarray``, adding two new attributes:
+
+* ``frame_no``, an integer
+* ``metadata``, a dictionary
+
+These can be used by the PIMS readers to provide any metadata stored in the
+image files. Setting these attriburtes is optional.
```
Comment (caspervdw) [inline doc/source/frame.rst]: attributes

File: doc/source/frame.rst [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (caspervdw) [inline doc/source/frame.rst]: For convenience, Frame objects in IPython notebook are displayed as actual images. 3D images ("Z stacks") are displayed as a stack of images. The user can scroll through the images with the scroll wheel.

File: doc/source/custom_readers.rst [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+       ...
+
+       @classmethod
+       def class_exts(cls):
+           return {'xyz'} | super(MyReader, cls).class_exts()
+
+New readers can be defined or imported at any time. They will be detected the
+next time ``pims.open`` is called, as it searches all subclasses of
+``FramesSequence`` for an eligible reader.
+
```
Comment (caspervdw) [inline doc/source/custom_readers.rst]: To prioritize readers, a field ``class_priority`` can optionally be given to the reader. A higher priority will be chosen over a lower priority. Default for all readers is `10`.

**Comment by caspervdw** [comment]
Great work! I read through them one time and left a couple of notes. After that it is ready for merge :+1:

**Comment by danielballan** [comment]
Thanks for the thorough read and good comments.

File: doc/source/custom_readers.rst [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    sheet : ndarray
+        The sprite sheet.  It should consist of N paneled images.  In
+        this version all possible positions have an image, this may be changed
+        to limit the number of acessable images in the sheet to be less than
+        the possible number.
+    rows : int
+    cols : int
+        The number of rows and columns of sprites.
+        The sprite size is computed from these + the shape of the sheet.
+    dtype : np.dtype or None
```
Comment (tacaswell) [inline doc/source/custom_readers.rst]: Copy-paste bug from me, this init function does not take a dtype

File: doc/source/image_sequence.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+An image sequence is a collection of image files with sequential
+filenames.
+
+``ImageSequence`` can be instatiated using:
+
+* a directory name, such as a ``'my_directory'``
+* a "glob" string, such as ``'my_directory/*.png'``, which is safer than
+  using a directory because directories sometimes contain stray files
+* the filepath of a zipped archive, such as ``'my_directory/all-images.zip'``
```
Comment (tacaswell) [inline doc/source/image_sequence.rst]: This can also take an explicit list of file paths.

File: doc/source/image_sequence.rst [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+Several widely-used Python packages have slightly different implementations
+of ``imread``, a general purpose image-reading function that understands
+popular formats like PNG, JPG, TIFF, and others. PIMS requires **one of
+the following** packages, in order of decreasing preference.
+
+* `scikit-image <http://scikit-image.org/>`_
+* `matplotlib <http://scikit-image.org/>`_
+* `scipy <http://scikit-image.org/>`_
+
+Scikit-image is included in the pims conda package.
```
Comment (tacaswell) [inline doc/source/image_sequence.rst]: included -> required by

File: doc/source/install.rst [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+source
+------
+
+If you plan to edit the code, you can install PIMS manually.
+
+.. code-block:: bash
+
+   git clone https://github.com/soft-matter/pims
+   pip install -e pims
```
Comment (tacaswell) [inline doc/source/install.rst]: I have never used pip for source install, I always do `python setup.py [develop|install]`

File: doc/source/tiff_stack.rst [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+``TiffStack_tifffile``, ``TiffStack_libtiff``, or ``TiffStack_pil``, which
+depend, respectively, on the packages below. The "default" reader,
+``TiffStack`` is an alias. At import time, it is pointed to the first
+reader for which the required package is installed.
+
+* `tifffile <https://pypi.python.org/pypi/tifffile>`_
+* `pylibtiff <https://pypi.python.org/pypi/libtiff>`_
+* `Pillow <https://pillow.readthedocs.org/>`_ or `PIL <http://www.pythonware.com/products/pil/>`_
+
+Tifffile is included in the pims conda package.
```
Comment (tacaswell) [inline doc/source/tiff_stack.rst]: included -> required by

**Comment by danielballan** [comment]
Addressed line notes.

**Pull request merged**